docs/add-ons/custom-addon.mdx

```diff
@@ -1,30 +1,43 @@
-# Overview
+# Custom Add-on
 
-Widgetbook allows developers to define their own Add-on. 
-Therefore, Widgetbook is fully customizable to your needs. 
+Widgetbook offers developers the flexibility to design their own Add-ons, making
+it fully adaptable to your requirements.
 
 <Warning>
-    As we are currently refining the API, the documentation is a bit sparse.
+  We are currently improving the API, and as a result, the documentation might
+  be brief.
 </Warning>
 
 ## Setup
 
-<Tabs
-  groupId="approach"
-  values={[
-    { label: 'Generator', value: 'generator' },
-    { label: 'Manual', value: 'manual' },
-  ]}
->
-  <TabItem value="generator">
-    Not yet supported, but coming soon.
-  </TabItem>
-  <TabItem value="manual">
-    Create a new class that extends the `WidgetbookAddOn` class with a generic parameter that is of type `WidgetbookAddOnModel`. 
-    Create the model that extends `WidgetbookAddOnModel`. 
-
-    <Warning>
-        More infos coming soon.
-    </Warning> 
-  </TabItem>
-</Tabs>
\ No newline at end of file
+To create a custom add-on, follow these steps:
+
+- Define a model for your custom add-on. This model will hold the data and
+  behavior of your add-on.
+- Create a new class that extends the WidgetbookAddon class with a generic
+  parameter corresponding to your defined model.
+
+Here's an example demonstrating how to create a custom add-on:
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+
+class CustomAddonModel {
+  // define the properties of your custom add-on model here
+}
+
+class CustomAddon extends WidgetbookAddon<CustomAddonModel> {
+  CustomAddon({
+    required CustomAddonModel initialSetting,
+  }) : super(
```

Comment:
Let's skip this for now; There are a lot to explain about the `Field` , so I have pushed a new example; check. This does not make sense. We need to write better examples and explanations about custom addons. 

One thing I noticed is that the Fields are not exposed `StringField` for example I cannot import this `StringField` or other ones. Look at the code that I pushed. it just doesn't work.

Reply:
> One thing I noticed is that the Fields are not exposed StringField for example I cannot import this StringField or other ones. Look at the code that I pushed. it just doesn't work.

@mhadaily We fixed that in #728 yesterday, and it will be released in `rc.2` soon. Please rebase your branch on the latest changes from `main` to get the latest changes.

Reply:
I agree that it can wait, as custom addons/knobs isn't a highly adopted feature.